independent-projects/arc/runtime/src/main/java/io/quarkus/arc/impl/InnerInvocationContext.java

```diff
@@ -0,0 +1,155 @@
+package io.quarkus.arc.impl;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Method;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import jakarta.interceptor.InvocationContext;
+
+import io.quarkus.arc.ArcInvocationContext;
+
+abstract class InnerInvocationContext implements ArcInvocationContext {
```

Comment:
I originally thought this is a generic superclass that just delegates to an underlying invocation context (in which case it could be just called `DelegatingInvocationContext`), but the `proceed()` contract seems to be pretty closely tied to the sole existing subclass. Maybe merge this and the subclass into one?

Reply:
OK, I see the next commit also has a class that extends from this. It doesn't seem to have an optimized variant for just one `@AroundInvoke` on the target class, likely because that isn't a very common thing, so they are pretty similar and the `proceed()` contract works for both of them.

I think I'd still call this a `DelegatingInvocationContext` and add some comments to the `proceed()` method, but that's just me :-)

But it makes sense now.

Reply:
I agree that it's not a good name but `DelegatingInvocationContext` is not good either. This IC is special because it does not proceed to other interceptors in the chain. So it's someting like an IC for a "local/inner invocation chain".

Reply:
Ah that's where the "inner" comes from! Well, I guess adding something like this would be enough then:

```java
/**
 * Invocation context for an "inner" invocation chain, consisting of interceptor methods declared
 * in one class and its superclasses. It doesn't proceed to other interceptors in the "outer" invocation
 * chain (interceptor methods declared in other classes).
 */

Reply:
Makes sense. Would you care to add a commit to this PR?

Reply:
I can do that, sure.

Reply:
Done.